### PR TITLE
fix Issue 24184 - [REG 2.103] Segmentation fault accessing variable w…

### PR DESCRIPTION
**Pull request by WalterBright**
…ith align(N) > platform stack alignment

Forgot to add an offset when the context pointer wasn't RPB.

### CODE REVIEW

File: compiler/src/dmd/e2ir.d [hunk 1]
```diff
@@ -609,7 +609,13 @@ elem* toElem(Expression e, ref IRState irs)
                     soffset = v.offset;
                 else if (v && v.inAlignSection)
                 {
-                    ethis = el_bin(OPadd, TYnptr, ethis, el_long(TYnptr, fd.salignSection.Soffset));
+                    ulong vthisOffset;
```
Comment (SixthDot) [inline compiler/src/dmd/e2ir.d:612]: can be `const`

```diff
-                    ulong vthisOffset;
+                    const vthisOffset = fd.vthis ? -toSymbol(fd.vthis).Soffset : 0;                                                    
-                    {
-                        Symbol *vs = toSymbol(fd.vthis);
-                        vthisOffset = -vs.Soffset;
-                    }
```

Comment (WalterBright) [reply compiler/src/dmd/e2ir.d]: ok

**Comment by SixthDot** [comment]
Recent regression, that should go on stable.

**Comment by WalterBright** [comment]
It's not really a regression, it never worked.

**Comment by ibuclaw** [comment]
Runtime corruption fixes still better in stable than waiting for the next release.

**Review by RazvanN7 — APPROVED** [review]
I rebased this on top of stable. All good to go.

**Comment by WalterBright** [comment]
Thanks @RazvanN7

**Pull request merged**